SimCore/src/SimCore/Generators/LHEPrimaryGenerator.cxx

```diff
@@ -1,53 +1,35 @@
-/**
- * @file LHEPrimaryGenerator.cxx
- * @brief Implementation file for LHEPrimaryGenerator
- * @author Jeremy McCormick, SLAC National Accelerator Laboratory
- * @author Tom Eichlersmith, University of Minnesota
- */
-
 #include "SimCore/Generators/LHEPrimaryGenerator.h"
 
-// Geant4
-#include "G4Event.hh"
-#include "G4IonTable.hh"
-#include "G4PhysicalConstants.hh"
-#include "G4RunManager.hh"
-#include "G4SystemOfUnits.hh"
-
-// LDMX
-#include "Framework/Configure/Parameters.h"
-#include "Framework/Exception/Exception.h"
-#include "SimCore/G4User/UserPrimaryParticleInformation.h"
-
 namespace simcore {
 namespace generators {
 
 LHEPrimaryGenerator::LHEPrimaryGenerator(
     const std::string& name, const framework::config::Parameters& parameters)
     : PrimaryGenerator(name, parameters) {
   file_path_ = parameters.getParameter<std::string>("filePath");
-  reader_ = new simcore::lhe::LHEReader(file_path_);
+  reader_ = std::make_unique<simcore::lhe::LHEReader>(file_path_);
 }
 
-LHEPrimaryGenerator::~LHEPrimaryGenerator() { delete reader_; }
-
 void LHEPrimaryGenerator::GeneratePrimaryVertex(G4Event* anEvent) {
-  simcore::lhe::LHEEvent* lheEvent = reader_->readNextEvent();
+  std::unique_ptr<simcore::lhe::LHEEvent> lheEvent = reader_->readNextEvent();
 
   if (lheEvent != nullptr) {
-    G4PrimaryVertex* vertex = new G4PrimaryVertex();
+    auto vertex = std::make_unique<G4PrimaryVertex>();
     vertex->SetPosition(lheEvent->getVertex()[0], lheEvent->getVertex()[1],
                         lheEvent->getVertex()[2]);
-    vertex->SetWeight(lheEvent->getXWGTUP());
+    vertex->SetWeight(lheEvent->getEventWeight());
 
     std::map<simcore::lhe::LHEParticle*, G4PrimaryParticle*> particleMap;
 
-    const std::vector<simcore::lhe::LHEParticle*>& particles =
-        lheEvent->getParticles();
-    for (auto* particle : particles) {
-      if (particle->getISTUP() > 0) {
-        G4PrimaryParticle* primary = new G4PrimaryParticle();
-        if (particle->getIDUP() == -623) { /* Tungsten ion */
+    const auto& particles = lheEvent->getParticles();
+    for (const auto& particle : particles) {
+      // Check if the particle has a valid, outgoing particle status
+      if (particle->getStatus() > 0) {
+        // Create a primary particle for the Geant4
+        auto primary = std::make_unique<G4PrimaryParticle>();
```

Comment:
Unfortunately, since Geant4 uses custom allocators, we can't manage them with STL smart pointers since there isn't really a way to move ownership*

Fortunately, Geant4 prefixes all of their classes with `G4` so we can follow the rule of thumb: `new` is okay only if its before `G4`.

*caveat: we could use [release](https://en.cppreference.com/w/cpp/memory/unique_ptr/release) like you do with the UserPrimaryParticleInformation, but I only think that's worth it if we want to impose some kind of `clang-tidy` rule that does not allow any `new`s.